nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java

```diff
@@ -287,41 +289,43 @@ public String toString() {
         return getMessage();
     }
 
+    private static class SourceDetector {
 
-    private void detectClassAndMethod(final LogRecord wrappedRecord) {
-        final StackTraceElement[] stack = Thread.currentThread().getStackTrace();
-        boolean ignoredClass = false;
-        for (final StackTraceElement element : stack) {
-            final String className = element.getClassName();
-            if (!ignoredClass) {
-                ignoredClass = isIgnoredStackTraceElement(className);
-                continue;
-            }
-            if (!isIgnoredStackTraceElement(className)) {
-                wrappedRecord.setSourceClassName(className);
-                wrappedRecord.setSourceMethodName(element.getMethodName());
-                return;
-            }
-        }
-    }
-
-
-    /**
-     * @param sourceClassName usually class which created this record
-     * @return if true the class will not be used as a source.
-     */
-    protected boolean isIgnoredStackTraceElement(final String sourceClassName) {
-        return "org.glassfish.main.jul.GlassFishLogger".equals(sourceClassName)
-            || "org.glassfish.main.jul.GlassFishLoggerWrapper".equals(sourceClassName)
+        private static final Set<String> IGNORED_CLASSES = Set.of(
+            "org.glassfish.main.jul.GlassFishLogger",
+            "org.glassfish.main.jul.GlassFishLoggerWrapper",
             // see LogDomains in GlassFish sources
-            || "com.sun.logging.LogDomainsLogger".equals(sourceClassName)
+            "com.sun.logging.LogDomainsLogger",
             // remaining classes are in the JDK
-            || "java.util.logging.Logger".equals(sourceClassName)
-            || "java.util.logging.LoggingProxyImpl".equals(sourceClassName)
+            "java.util.logging.Logger",
+            "java.util.logging.LoggingProxyImpl",
             // see LoggingPrintStream
-            || "java.lang.Throwable".equals(sourceClassName)
-            || sourceClassName.startsWith("java.lang.reflect.")
-            || sourceClassName.startsWith("sun.util.logging.")
-            || sourceClassName.startsWith("sun.reflect.");
+            "java.lang.Throwable"
+        );
+
+        private static final StackWalker STACK_WALKER = StackWalker.getInstance(StackWalker.Option.SHOW_REFLECT_FRAMES);
+
+        static void detectClassAndMethod(final LogRecord wrappedRecord) {
+            STACK_WALKER
+                .walk(stackFrames ->
+                    stackFrames.dropWhile(frame -> !isIgnoredStackFrame(frame.getClassName()))
```

Comment:
Just for fun - if there only could be `isNotIgnoredStackFrame(StackWalker.StackFrame frame)` method (`::`-referenced in both `dropWhile` and `filter`)...

Reply:
@avpinchuk @pzygielo Can you do that? It is a good idea, unfortunately @arjantijms already merged the PR, so you have to create new one.

Reply:
Sure. But as a separate PR ))